apps/desktop/src/vault/app/vault/credential-generator-dialog.component.ts

```diff
@@ -46,8 +46,13 @@ export class CredentialGeneratorDialogComponent {
     private dialogService: DialogService,
   ) {}
 
-  algorithm = (selected: AlgorithmInfo) => {
-    this.buttonLabel = selected.useGeneratedValue;
+  onAlgorithmSelected = (selected?: AlgorithmInfo) => {
+    if (selected) {
+      this.buttonLabel = selected.useGeneratedValue;
+    } else {
+      // clear the credential value if we don't currently have an algorithm
+      this.credentialValue = undefined;
```

Comment:
👍🏻 It's a great idea to have a comment that explains why you're clearing the credentialValue!

🎨 The "why" is a bit more clear if it's stated in terms of user flow

```suggestion
      // clear the credential value when the user is
      // selecting the credential generation algorithm
      this.credentialValue = undefined;
```